### PR TITLE
Removing JSErrors

### PR DESCRIPTION
**Pull request by ssivov**

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import getPlugin from "./plugins";
 import { debug, getCookie, guid, isNumber, mapProperties, setCookie } from "./utils";
 
 // Constants
-const version = "0.1.7";
+const version = "0.1.8";
```
Comment (toby-walker) [inline src/core.ts]: Minor: Duplicates number in package.json. Any way to get them from one place?

File: src/core.ts [hunk 2]
```diff
@@ -117,8 +117,8 @@ export function instrument(eventState: IInstrumentationEventState) {
 }
 
 function getUnixTimestamp(): number {
-  return (window.performance && window.performance.now)
```
Comment (toby-walker) [inline src/core.ts]: Should you not check for the presence of windows.timing as well?

File: src/plugins/layout.ts [hunk 3]
```diff
@@ -96,40 +96,42 @@ export default class Layout implements IPlugin {
...
-            case Node.TEXT_NODE:
-              (layoutState as ITextLayoutState).content = originalProperties.characterData;
-              break;
-            case Node.ELEMENT_NODE:
-              let originalAttributes = (layoutState as IElementLayoutState).attributes;
-              for (let attr in originalAttributes) {
-                if (originalAttributes.hasOwnProperty(attr)) {
-                  (layoutState as IElementLayoutState).attributes[attr] = originalAttributes[attr];
+
+      if (shadowNode) {
```
Comment (toby-walker) [inline src/plugins/layout.ts]: When can this be null?

File: src/utils.ts [hunk 4]
```diff
@@ -6,8 +6,9 @@ import { instrument } from "./core";
 // tslint:disable
 export function guid() {
   let d = new Date().getTime();
-  if (window.performance && window.performance.now) {
```
Comment (toby-walker) [inline src/utils.ts]: Minor. You might want to put this in a function as it's duplicated in several places.

File: src/plugins/layout.ts (hunk 3 above)
Comment (ssivov) [inline src/plugins/layout.ts]: In design - never, however it is sometimes. I'm on a mission to find out when this happens as my next task.

File: src/core.ts (hunk 2 above)
Comment (ssivov) [inline src/core.ts]: There are some cases when performance object exists, without the 'now' method, but it always has the 'timing' property

File: src/core.ts (hunk 1 above)
Comment (ssivov) [inline src/core.ts]: Unfortunately, no

File: src/plugins/layout.ts (hunk 3 above)
Comment (ender336) [inline src/plugins/layout.ts]: We should explicitly log something in an else clause then so we have a way to investigate further

---
In reply to: [124159155](https://github.com/Microsoft/clarity-js/pull/20#discussion_r124159155) [](ancestors = 124159155)

**Pull request merged**